### ISSUE TITLE
Create a selection from a selection in the vector attribute table

### ISSUE THREAD

**Issue by lajunek** [qgis/QGIS#38154]
**Possibility to click on a selection in the attribute table and create a selection from a selection by clicking**

**Additional context**

**Comment by m-kuhn** [comment]
Hi @lajunek 
I am not completely sure what this feature request is for, can you add some more explanation? Thank you.

**Comment by uclaros** [comment]
I'd guess something like the highlight feature of arc. Select some rows, then highlight some of the selected, then click select/deselect highlighted. Right now it's quite easy to lose your selection by clicking the row header.

**Comment by baswein** [comment]
In case you aren't aware of it the new in 3.14  reselect features function under Edit>Selection>Reselect Features may ease some of your pain.

**Comment by uclaros** [comment]
Reselect was supposed to work against clear selection, not selection change. Has this changed?

**Comment by baswein** [comment]
I was mistaken- It only works against clear selection. Is there a reason selection isn't part of undo redo?

**Comment by lajunek** [comment]
@m-kuhn in detail: you select some vector features by holding strg. Then you click into the marked features and select features from the selection. The first selection is still there and doesn't dissappear. So that both selections remain in the "cache" of QGIS. Actually I'd like to have this feature in the QGIS 3.10 LTR because we only use LTR versions.

**Comment by m-kuhn** [comment]
Thanks @lajunek it might be just me standing on the wire but I am still having a hard time understanding. In particular I find it hard to follow "and select features from the selection". If the features are already in the selection, how can they be selected again?

**Comment by uclaros** [comment]
In arcmap you get two different types of selection. One is the normal selection just like in QGIS, and then once you have selected features you get a second level of selection within selected features called a highlight or something marking rows with a yellow color. After you have highlighted some selected features you can choose to remove them from the normal selection or to remove the non-highlighted from the selection. Maybe you could also zoom/pan to highlighted too, not sure, it's been years since I last used arcmap.

**Comment by m-kuhn** [comment]
So the "real" requirement is to be able to subtract some elements from an existing selection (and the subselection is an intermediate step there)?

**Comment by uclaros** [comment]
@m-kuhn  I'm not quite sure, since this can be easily handled with the ctrl key. I think the advantage of the proposed method (apart from being more friendly to arcmap users) is the visual interaction with the selected rows (features are also highlighted in the map) without the need to pan/zoom/flash to the feature explicitly, especially in *show selected features* mode.
I used to miss that arcmap feature but not any more.

@lajunek That's not how things work, 3.10 LTR receives bug fixes, not features. Even if this feature is implemented, it will end up in an LTR when 3.16.4 becomes LTR according to the schedule.

**Comment by lajunek** [comment]
@m-kuhn yes the real requirement is to be able to substract elements from an existing selection. It would be great to have such a feature, as my colleagues often use this feature (-: and will use 3.16.4 LTR.

**Comment by uclaros** [comment]
You *can* subtract elements from a selection by holding Ctrl and clicking on a selected row.

**Comment by m-kuhn** [comment]
I just stumbled upon this stackoverflow question and answer which might be of interest here: https://gis.stackexchange.com/a/378482/9839
